fuzzing/valuegeneration/mutator_shrinking.go

```diff
@@ -0,0 +1,207 @@
+var bytesShrinkingMethods = []func(*ShrinkingValueMutator, []byte) []byte{
+	// Replace a random index with a zero byte
+	func(g *ShrinkingValueMutator, b []byte) []byte {
+		if len(b) > 0 {
+			b[g.randomProvider.Intn(len(b))] = 0
+		}
+		return b
+	},
+	// Remove a random byte
+	func(g *ShrinkingValueMutator, b []byte) []byte {
+		// If we have no bytes to remove, do nothing.
+		if len(b) == 0 {
+			return b
+		}
+
+		i := g.randomProvider.Intn(len(b))
+		return append(b[:i], b[i+1:]...)
+	},
+}
+
+// MutateBytes takes a dynamic-sized byte array input and returns a mutated value based off the input.
+func (g *ShrinkingValueMutator) MutateBytes(b []byte) []byte {
+	randomGeneratorDecision := g.randomProvider.Float32()
+	if randomGeneratorDecision < g.config.ShrinkValueProbability {
+		// Mutate the data for our desired number of rounds
+		input := bytesShrinkingMethods[g.randomProvider.Intn(len(bytesShrinkingMethods))](g, b)
+		return input
+	}
+	return b
+}
+
+// integerShrinkingMethods define methods which take a big integer and a set of inputs and
+// transform the integer with a random input and operation. This is used in a loop to create
+// mutated integer values.
+var integerShrinkingMethods = []func(*ShrinkingValueMutator, *big.Int, ...*big.Int) *big.Int{
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		// Subtract a random input
+		var r *big.Int
+		if x.Cmp(big.NewInt(0)) > 0 {
+			r = big.NewInt(0).Sub(x, inputs[g.randomProvider.Intn(len(inputs))])
+		} else if x.Cmp(big.NewInt(0)) < 0 {
+			r = big.NewInt(0).Add(x, inputs[g.randomProvider.Intn(len(inputs))])
+		} else {
+			r = big.NewInt(0)
+		}
+		return r
+
+	},
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		// Divide by two
+		return big.NewInt(0).Div(x, big.NewInt(2))
+	},
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		return big.NewInt(0)
+	},
+}
+
+// MutateInteger takes an integer input and applies optional mutations to the provided value.
+// Returns an optionally mutated copy of the input.
+func (g *ShrinkingValueMutator) MutateInteger(i *big.Int, signed bool, bitLength int) *big.Int {
+	randomGeneratorDecision := g.randomProvider.Float32()
+	if randomGeneratorDecision < g.config.ShrinkValueProbability {
+		// Calculate our integer bounds
+		min, max := utils.GetIntegerConstraints(signed, bitLength)
+
+		// Obtain our inputs. We also add our min/max values for this range to the list of inputs.
+		// Note: We exclude min being added if we're requesting an unsigned integer, as zero is already
+		// in our set, and we don't want duplicates.
+		var inputs []*big.Int
+		inputs = append(inputs, g.valueSet.Integers()...)
+		if signed {
+			inputs = append(inputs, min, max)
+		} else {
+			inputs = append(inputs, max)
+		}
+
+		// Determine which value we'll use as an initial input, and how many mutations we will perform.
```

Comment:
Remove "how many mutations we will perform" since we are not doing rounds of shrinking

Reply:
Updated in https://github.com/crytic/medusa/pull/147/commits/c89252203c403f26938e1375af059451a3469d77